src/lib/contact.ts

```diff
@@ -45,6 +46,47 @@ class Contact {
 		});
 	}
 
+	public async updateProgramName(
+		concernedPrograms: IPCProgram,
+		newName: string,
+		sharedPrograms: IPCProgram[]
+	  ): Promise<ResponseType> {
+		try {
+		  let programFound = false;
+		  await Promise.all(
+			this.contacts.map(async (contact) => {
+			  const program = contact.programs.find((f) => f.id === concernedPrograms.id);
+			  if (program) {
+				program.name = newName;
+				program.logs.push({
+				  action: `Renamed program to ${newName}`,
+				  date: Date.now(),
+				});
+				programFound = true;
+				await this.publishAggregate();
+			  }
+			})
+		  );
+		  if (!programFound) {
+			const program = sharedPrograms.find((f) => f.id === concernedPrograms.id);
+			if (!program) {
+			  return { success: false, message: 'Program not found' };
+			}
+			await post.Publish({
+			  account: this.account,
+			  postType: 'InterPlanetaryCloud',
+			  content: { program: { ...concernedPrograms, name: newName }, tags: ['yolo', concernedPrograms.id] },
```

Comment:
I think the key is `programs` (with an `s`). You added tags, is there a reason for this ? Do you understand what they are used for ?

Reply:
If I understood the tags are here to find the content easily right? if no can you please explain me this part.